src/snowcli/app/loggers.py

```diff
@@ -1,32 +1,132 @@
+            logging.INFO,
+            logging.WARN,
+            logging.ERROR,
+            logging.CRITICAL,
+        ]
+        if self.file_log_level not in possible_log_levels:
+            raise InvalidLogsConfiguration(
+                f"Invalid 'level' value set in [logs] section: {config['level']}. "
+                f"'level' should be one of: {' / '.join(logging.getLevelName(lvl) for lvl in possible_log_levels)}"
+            )
+
+    @property
+    def filename(self):
+        return self.path / _DEFAULT_LOG_FILENAME
 
 
 def create_loggers(verbose: bool, debug: bool):
-    """Creates a logger depending on the SnowCLI parameters
+    """Creates a logger depending on the SnowCLI parameters and config file.
     verbose == True - print info and higher logs in default format
     debug == True - print debug and higher logs in debug format
     none of above - print only error logs in default format
     """
-    logger = logging.getLogger("snowcli")
+    config = LogsConfig(debug=debug)
+
     if verbose and debug:
         raise typer.BadParameter("Only one parameter `verbose` or `debug` is possible")
     elif debug:
-        log_format = DEBUG_LOG_FORMAT
-        log_level = logging.DEBUG
+        console_log_format = DEBUG_LOG_FORMAT
+        console_log_level = logging.DEBUG
     elif verbose:
-        log_format = DEFAULT_LOG_FORMAT
-        log_level = logging.INFO
+        console_log_format = DEFAULT_LOG_FORMAT
+        console_log_level = logging.INFO
+    else:
+        console_log_format = DEFAULT_LOG_FORMAT
+        console_log_level = logging.ERROR
+
+    if console_log_level < config.file_log_level:
+        global_log_level = console_log_level
     else:
-        log_format = DEFAULT_LOG_FORMAT
-        log_level = logging.ERROR
-
-    console = logging.StreamHandler()
-    console.setFormatter(logging.Formatter(log_format, "%Y-%m-%d %H:%M:%S"))
-    logger.addHandler(console)
-    logger.setLevel(log_level)
-    logging.getLogger("snowflake").setLevel(log_level)
+        global_log_level = config.file_log_level
+
+    enabled_handlers = ["console"]
+    handlers_config = {
+        "console": {
+            "class": "logging.StreamHandler",
+            "formatter": "console_formatter",
+            "level": console_log_level,
+        },
+    }
+    if config.save_logs:
+        enabled_handlers.append("file")
+        handlers_config["file"] = {
+            "class": "logging.handlers.TimedRotatingFileHandler",
+            "filename": config.filename,
+            "when": "midnight",
+            "formatter": "file_formatter",
+            "level": config.file_log_level,
+        }
+
+    logging.config.dictConfig(
+        {
+            "version": 1,
+            "disable_existing_loggers": True,
+            "formatters": {
+                "console_formatter": {
+                    "class": "logging.Formatter",
+                    "format": console_log_format,
+                    "datefmt": DATE_FORMAT,
```

Comment:
Do we need to specify it? Won't this work out of box?

Reply:
It would, but it would give you default formatter:
```
In [3]: logging.error("message")
ERROR:root:message
```
If we want to use "prettier" format, we need to override it here